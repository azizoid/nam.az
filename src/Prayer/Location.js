import React from "react";
import Clock from "./Clock";
import { FaChevronLeft, FaChevronRight } from "react-icons/fa";

<<<<<<< Updated upstream
const Location = ({ location, tarix, hijri, doy, changeDoy }) => {
  return (
    <div className="d-flex align-items-center justify-content-center">
      <button
        className="btn btn-link"
        style={{ fontSize: "3em", color: "#6cb2eb" }}
        onClick={() => changeDoy(doy - 1)}
      >
        <FaChevronLeft />
      </button>
      <div className="text-center col-md-5" id="location">
=======
const Location = ({ location, tarix, doy, changeDoy }) => {
  return (
    <div className="d-flex justify-content-center">
      <button
        className="btn btn-link"
        style={{ fontSize: "3em", color: "#6cb2eb" }}
        onClick={() => {
          changeDoy(doy - 1);
        }}
      >
        <FaChevronLeft />
      </button>
      <div className="text-center d-none d-md-block" id="location">
>>>>>>> Stashed changes
        <h1 className="nowis">
          <Clock />
        </h1>
        <h1>{location}</h1>
        <small>{tarix}</small>
<<<<<<< Updated upstream
        <br />
        <small>{hijri}</small>
=======
>>>>>>> Stashed changes
      </div>
      <button
        className="btn btn-link"
        style={{ fontSize: "3em", color: "#6cb2eb" }}
<<<<<<< Updated upstream
        onClick={() => changeDoy(doy + 1)}
=======
        onClick={(e) => {
          changeDoy(doy + 1);
        }}
>>>>>>> Stashed changes
      >
        <FaChevronRight />
      </button>
    </div>
  );
};

export default Location;
