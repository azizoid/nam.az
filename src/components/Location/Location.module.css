.locationNavBtn {
    font-size: 3em;
    color: #6cb2eb 
}