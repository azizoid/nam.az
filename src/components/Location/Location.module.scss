.locationNavBtn {
  font-size: 3em;
  color: #6cb2eb;
}

.location {
  padding-top: 3rem;
  padding-bottom: 1rem;

  h2 {
    font-weight: bold;
    color: #4a4a4a;

    &.nowis {
      color: lightgrey;
    }
  }
}

@media (max-width: 480px) {
  .location {
    padding: 0rem;
  }
}
