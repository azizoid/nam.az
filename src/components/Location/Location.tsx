import moment from 'moment-hijri';
moment.locale('az');
import { MdNavigateBefore, MdNavigateNext } from 'react-icons/md';
import { hijriMonthList } from '../../assist/hijriMonthList';
import { numberSuffixAz } from '../../utility/numberSuffixAz/numberSuffixAz';

import { Clock } from './Clock/Clock';

export type LocationProps = {
  location: string;
  tarix: string;
  hijri: string;
  dd: number;
  changeDd: (dd: number) => void;
};

const Location = ({
  location,
  tarix,
  dd,
  changeDd,
}: LocationProps): JSX.Element => {
  const momentTarix = moment(tarix);

  return (
    <div className="flex justify-around align-middle">
      <button className="btn text-blue-300" onClick={() => changeDd(dd - 1)}>
        <MdNavigateBefore />
      </button>

      <div className="flex flex-col space-y-2 text-center text-3xl font-semibold">
        <h2 className="text-slate-300">
          <Clock />
        </h2>

        <h2>{location}</h2>

        <small className="text-sm font-normal">
          {hijriMonthList[Number(momentTarix.format('iM')) - 1]} ayı{', '}
          {numberSuffixAz(Number(momentTarix.format('iD')))} gün{', '}
          {numberSuffixAz(Number(momentTarix.format('iYYYY')))} il
          <br />
          {tarix}
        </small>
      </div>

      <button className="btn text-blue-300" onClick={() => changeDd(dd + 1)}>
        <MdNavigateNext />
      </button>
    </div>
  );
};

export default Location;
