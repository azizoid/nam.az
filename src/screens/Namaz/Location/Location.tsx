import { MdNavigateBefore, MdNavigateNext } from 'react-icons/md';

import { Clock } from './Clock/Clock';
import Link from 'next/link';

export type LocationProps = {
  city: number;
  location: string;
  tarix: string;
  hijri?: string;
  dayOfYear: number;
};

export const Location = ({
  city,
  location,
  tarix,
  hijri,
  dayOfYear,
}: LocationProps) => {
  console.log(dayOfYear)
  return (
    <div className="flex justify-around align-middle">
      <Link href={`/${city}/${dayOfYear - 1}`} className="btn text-blue-300">
        <MdNavigateBefore />
      </Link>
      {/* <button className="btn text-blue-300" onClick={() => navigate(dayOfYear - 1)}>
        <MdNavigateBefore />
      </button> */}

      <div className="flex flex-col space-y-2 text-center text-3xl font-semibold">
        <h2 className="text-slate-300">
          <Clock />
        </h2>

        <h2>{location}</h2>

        <small className="text-sm font-normal">
          {tarix}, <span className="whitespace-nowrap">{hijri}</span>
        </small>
      </div>

      {/* <button className="btn text-blue-300" onClick={() => navigate(dayOfYear + 1)}>
        <MdNavigateNext />
      </button> */}
      <Link href={`/${city}/${dayOfYear + 2}`} className="btn text-blue-300">
        <MdNavigateNext />
      </Link>
    </div >
  )
};
