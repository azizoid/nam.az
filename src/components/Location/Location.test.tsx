import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import MockDate from 'mockdate';

import Location from './Location';

const mockDate = Date.parse('2021-12-06T01:14:36.710Z');

MockDate.set(mockDate);

const mockProps = {
  location: 'Bakı',
  tarix: 'Şənbə, 2 aprel 2022',
  dd: 93,
  changeDd: jest.fn(),
};

test('renders Location', async () => {
  const { container, getByText } = render(<Location {...mockProps} />);

  const city = getByText('Bakı');

  expect(city).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
