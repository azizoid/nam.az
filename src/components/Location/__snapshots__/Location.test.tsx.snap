// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Location 1`] = `
<div>
  <div
    class="flex justify-around align-middle"
  >
    <button
      class="btn text-blue-300"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </button>
    <div
      class="flex flex-col space-y-2 text-center text-3xl font-semibold"
    >
      <h2
        class="text-slate-300"
      >
        <p
          class="clock"
        >
          05:14:36
        </p>
      </h2>
      <h2>
        Bakı
      </h2>
      <small
        class="text-sm font-normal"
      >
        Ramazan
         ayı
        , 
        1-ci
         gün
        , 
        1443-cü
         il
        <br />
        Şənbə, 2 aprel 2022
      </small>
    </div>
    <button
      class="btn text-blue-300"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
